src/box/lua/tuple.lua

```diff
@@ -65,6 +67,32 @@ local builtin = ffi.C
 local tuple_t = ffi.typeof('box_tuple_t')
 local const_tuple_ref_t = ffi.typeof('box_tuple_t&')
 
+local new_tuple = function(...)
+    if compat.box_tuple_new_arg_list.current == 'old' then
+        return box.tuple._new(...)
+    end
+    local args = {...}
```

Comment:
Seems here you can use

select('#', ...) to get varargs length and select(N, ...) to get Nth element without excess table creation.

Reply:
Thanks for the suggestion! Applied it.